SOC-1446 Add edit token for forum admin tools

armonr opened the pull request:
/CC @Grunny @garthwebb @kvas-damian

Commit:
SOC-1446 Add edit token for forum admin tools

Grunny commented on extensions/wikia/Forum/Forum.class.php:
```diff
@@ -238,6 +238,12 @@ protected function createOrEditBoard( $board, $titletext, $body, $bot = false )
 		}
 
+		if ( !F::app()->wg->request->wasPosted() ||
+			!F::app()->wg->User->matchEditToken( F::app()->wg->request->getVal( 'editToken' ) ) ) {
```
These can use `$this->wg` to access the globals, rather than `F::app`.

Grunny commented on extensions/wikia/Forum/ForumSpecialController.class.php:
```diff
@@ -140,6 +140,11 @@ public function createNewBoardModal() {
+				array(
+					'type' => 'hidden',
+					'name' => 'token',
+					'value' => F::app()->wg->User->getEditToken(),
```
`$this->getUser()->getEditToken()` would be preferred here and below.

Grunny commented:
I can see how this simplifies the checks, but in general, I think it would be better if the controllers were responsible for checking the validity of requests (which could be done by using [`$this->checkWriteRequest()`](https://github.com/Wikia/app/blob/3c714f0e8d3/includes/wikia/nirvana/WikiaDispatchableObject.class.php#L189-L202) to them), just as the ForumExternalController is also responsible for checking permissions right now through `checkAdminAccess`.

kvas-damian commented:
In addition to what @Grunny said I think we should introduce new error message for expired or not valid edit token. Current solution for this kind of errors displays `forum-board-title-validation-invalid` when trying to create new board.

Commit:
Add an new message code in forum,
reuses https://github.com/Wikia/app/blob/dev/extensions/wikia/WikiaQuiz/WikiaQuiz.i18n.php#L62-62

Commit:
Check token on controller actions & extend to remove action

armonr commented:
Thanks @Grunny and @kvas-damian. Comments were addressed.

garthwebb commented on extensions/wikia/Forum/ForumExternalController.class.php:
```diff
@@ -58,6 +58,12 @@ public function createNewBoard() {
 		}
 
+		try {
+			$this->displayAnyWriteAccessError();
```
Two things.  First, I think functions like this should be named for what they intend to do, not what they might do in an exception.  In this case, that would be something like `checkEditToken()`.  Second, given a function that does this already exists as `checkWriteRequest()` why not call that here rather than define another function that just adds another layer of `try`/`catch` blocks?  If you want to reduce repetition, I'd make a method that just sets the return values for you and use it here, e.g.:

``` php
function setMismatchError() {
    $this->status = 'error';
    $this->errormsg = wfMessage( 'forum-token-mismatch' )->escaped();
}
...
try {
    $this->checkWriteRequest();
} catch ( \BadRequestException $bre ) {
    $this->setMismatchError();
    return; # no need to return a boolean here; Oasis doesn't care what controllers return
}
```

Commit:
Update token mismatch handling

kvas-damian commented on extensions/wikia/Forum/ForumExternalController.class.php:
```diff
@@ -161,6 +175,12 @@ public function removeBoard() {
 		}
 
+		try {
+			$this->setTokenMismatchError();
```
as @garthwebb already mentioned somewhere: something is wrong here.

armonr replied:
o.O Somehow missed the commit train! Will fix!

Commit:
SOC-1446 Fix last instance of handling token matching

garthwebb commented:
LGTM
